tools/tests/test_nd_freeze

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+# Temporary scratch directory to save generated files.
+TMP_DIR="tmp_$((1 + RANDOM % 10000))"
+
+# Run common commands for each test at start of tests
+#
+# Parameters
+# ----------
+# distro : string
+#   Valid values are "debain" or "ubuntu"
+# release : string
+#   Code name for distro release. (e.g. jessie)
+# date : string
+#   Date at which the snapshot of the OS is wanted
+# keep_sources : string (Optional)
+#   Pass the string "KEEP_SOURCES" to run nd_freeze with the --keep-original-apt-sources switch
+#
+function test_setup {
+    distro=$1
+    release=$2
+    date=$3
+    keep_sources=${4:-no}
+
+    if [ "$keep_sources" = "no" ]; then
+        keep_sources_param=""
+    else
+        keep_sources_param="--keep-original-apt-sources"
+    fi
+
+    mkdir -p $TMP_DIR
+
+    docker run -it --rm -v $PWD/..:/home ${distro}:${release} /bin/bash -c "
+        apt-get update
+        apt-get install -y wget
+        wget -O- http://neuro.debian.net/lists/${release}.us-nh.full | tee /etc/apt/sources.list.d/neurodebian.sources.list
+        apt-key adv --recv-keys --keyserver hkp://pool.sks-keyservers.net:80 0xA5D32F012649A5A9
```

Comment:
why not just use the neurodebian images to avoid all of the above? we should have them for each of debian/ubuntu releases: https://hub.docker.com/_/neurodebian/